src/util/emitUtils.js

```diff
@@ -355,20 +359,44 @@ function initializeBotNamespace(inst, io) {
  * @param {String} namespace - The namespace to initialize
  * @param {Socket.io} io - The socketio's server side object
  */
-function initializeNamespace(namespace, io) {
+function initializeNamespace(namespace, io, processName) {
   connectedRooms[namespace] = new Set();
   io.of(namespace).on('connect', (socket) =>
     Promise.join(
       validateIp(socket),
       validateTokenNewFormat(socket),
     )
-    .then(() => {
+    .then((responses) => {
+      const userInfo = {
+        ipAddress: responses[0],
+      };
+
+      /*
+       * If we got back a "token" object (i.e. has property "isRevoked") then
+       * this request was initiated via API. If we got back a "user" object
+       * then this request was initiated via UI (user name = token name).
+       */
+      if (responses[1].body.hasOwnProperty('isRevoked')) {
+        userInfo.name = responses[1].body.user.name;
+        userInfo.tokenName = responses[1].body.name;
+      } else {
+        userInfo.name = responses[1].body.name;
+        userInfo.tokenName = responses[1].body.name;
+      }
+
+      if (toggle.isFeatureEnabled('activityLoggingUser')) {
+        logger.info(`activity=user:connect ipAddress=${userInfo.ipAddress} ` +
+          `nsp=${socket.nsp.name} process=${processName} ` +
+          `token=${userInfo.tokenName} user=${userInfo.name}`);
+      }
+
       addToRoom(socket);
-      trackConnectedRooms(socket);
+      trackConnectedRooms(socket, userInfo, processName);
       socket.emit('authenticated');
     })
     .catch((err) => {
       pubSubStats.trackAuthError();
+      logger.error('auth error', err.message);
```

Comment:
We had a discussion about this in the token auth PR, with you arguing against logging auth errors. Why did you change your mind?

Reply:
You are absolutely right. Now that we're looking at actual real-time traffic, I find I do want to see a little more detail on the auth errors... mostly to make sure they *are* indeed expected.